### PR TITLE
feat(sections): add description to add-sections and update-sections

### PR DESCRIPTION
**Pull request by lmjabreu**
## What

Surfaces section descriptions (26Q2 Project Essentials) in the MCP tools, matching the CLI surface.

- `add-sections` accepts an optional `description` parameter
- `update-sections` accepts `description` and makes `name` optional, so description-only updates work
- the section output schema and `toSectionSummary` now include `description`

## ⚠️ SDK dependency (blocks the read side)

`@doist/todoist-sdk` (pinned 10.3.3) does **not** model section descriptions yet, even though the backend shipped them on 2026-06-10:

- the SDK's `Section` type omits `description`, and the SDK strips it from section reads, so it's read through a widened type and the update args are cast
- the **write path works at runtime** (the REST client forwards the field), but the output `description` stays `undefined` until the SDK adds it to the section read schema and args

Opening as a draft for that reason. Needs an SDK release (add `description` to the section read schema; add `description` + optional `name` to `UpdateSectionArgs`).

## Same contract

Mirrors the project/CLI surface: an optional `description` parameter, output-schema field, and the typed escape hatch confined to the SDK gap. Output schema uses `z.string().optional()` (no `.nullable()`) to stay Gemini-compatible per the repo's schema linter.

## Scope

- **In scope:** section descriptions across add/update and the section output schema.
- **Non-goals:** project descriptions (separate PR, #500).

## Testing

- New tests assert `description` is forwarded to the API in `add-sections`, mapped into output, and that `update-sections` sends a description-only update with no `name`.
- Full suite green (913 tests), type-check, lint/format, schema validation, and build all pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by lmjabreu** [comment]
@Doistbot /review

File: src/tools/__tests__/update-sections.test.ts [hunk 1]
```diff
@@ -66,6 +66,24 @@ describe(`${UPDATE_SECTIONS} tool`, () => {
                 }),
             )
         })
+
+        it('should update description only, without a name change', async () => {
```
Comment (doistbot) [inline src/tools/__tests__/update-sections.test.ts:72]: <big><kbd>🟡 P2</kbd></big>  Per `AGENTS.md` testing requirements, new optional tool parameters must include test coverage for clearing values. The schema sets the contract `Pass "" to clear it.`, so there should be a test verifying an empty string is correctly mapped and forwarded in the API payload.

File: src/tools/update-sections.ts [hunk 2]
```diff
@@ -6,7 +6,11 @@ import { ToolNames } from '../utils/tool-names.js'
 
 const SectionUpdateSchema = z.object({
     id: z.string().min(1).describe('The ID of the section to update.'),
-    name: z.string().min(1).describe('The new name of the section.'),
+    name: z.string().min(1).optional().describe('The new name of the section.'),
+    description: z
+        .string()
+        .optional()
+        .describe('The description of the section. Supports Markdown. Pass "" to clear it.'),
```
Comment (doistbot) [inline src/tools/update-sections.ts]: <big><kbd>🟡 P2</kbd></big>  This adds a clearable optional field, but it doesn't follow the repo's established clearing contract. A caller that sends `description: "remove"` (or legacy `null`) will have that value forwarded verbatim instead of clearing the description. Please accept the standard clear sentinel in the schema and translate it to `null` before calling `updateSection`.

**Review by doistbot — COMMENTED** [review]
This PR successfully surfaces section descriptions in the add and update tools while neatly working around the current SDK typing gap.

Few things worth tightening:
- **Clearing contract**: Use the repository's standard clearing sentinel (like `"remove"`) instead of an empty string to clear the description, translating it to `null` before making the API call.
- **Test coverage**: Add a test verifying this field-clearing behavior, aligning with the repository guidelines for new optional fields.

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-mcp&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-mcp%2Fpull%2F501&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253A7e9209c0-6513-11f1-9de0-7b494e47acf4%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3A7e9209c0-6513-11f1-9de0-7b494e47acf4&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/tools/update-sections.ts (hunk 2 above)
Comment (lmjabreu) [reply src/tools/update-sections.ts]: Done. Switched to the `"remove"` sentinel (handling legacy `null` via preprocess), mapping it to `null` (the section clear value) before the call — matching update-goals/update-tasks.

File: src/tools/__tests__/update-sections.test.ts (hunk 1 above)
Comment (lmjabreu) [reply src/tools/__tests__/update-sections.test.ts:72]: Done. Added tests for clearing via `"remove"` and via legacy `null` (parsed through the schema preprocess).

**Comment by lmjabreu** [comment]
@Doistbot /review

File: src/tools/update-sections.ts [hunk 3]
```diff
@@ -4,11 +4,29 @@ import type { TodoistTool } from '../todoist-tool.js'
 import { SectionSchema as SectionOutputSchema, toSectionSummary } from '../utils/output-schemas.js'
 import { ToolNames } from '../utils/tool-names.js'
 
+const REMOVE_SENTINEL = 'remove'
+
 const SectionUpdateSchema = z.object({
     id: z.string().min(1).describe('The ID of the section to update.'),
-    name: z.string().min(1).describe('The new name of the section.'),
+    name: z.string().min(1).optional().describe('The new name of the section.'),
```
Comment (doistbot) [inline src/tools/update-sections.ts]: <big><kbd>🟡 P2</kbd></big>  Making `name` optional here also makes `{ sections: [{ id: "sec-1" }] }` a valid input, and `execute()` will then call `client.updateSection(id, {})`. Because that request is cast past the SDK's required-name type, callers now get a runtime no-op/invalid update instead of a validation error. Add a schema refine (or skip logic like the other update tools) so each section includes at least one of `name` or `description`.

File: src/tools/__tests__/update-sections.test.ts [hunk 4]
```diff
@@ -66,6 +67,56 @@ describe(`${UPDATE_SECTIONS} tool`, () => {
...
```
Comment (doistbot) [inline src/tools/__tests__/update-sections.test.ts:72]: <big><kbd>🟡 P2</kbd></big>  Now that `name` is optional, the suite should also cover the no-op input `{ id }` with neither `name` nor `description`. That’s the main regression path introduced by this change, and without a test it’s easy to miss that the tool currently falls through to `updateSection(id, {})`.

**Review by doistbot — COMMENTED** [review]
This PR surfaces section descriptions in the add and update tools while correctly working around the current SDK limitations.

Few things worth tightening:
- Add a schema refinement or skip logic to `update-sections` to prevent empty `{ id }` updates (lacking both `name` and `description`) from hitting the API.
- Include test coverage for this no-op `{ id }` update regression path.

I also included a few optional follow-up notes in the details below.

<details>
<summary>Optional follow-up notes (2)</summary>

- **[P3] src/tools/update-sections.ts:25**: This helper duplicates identical logic from `update-goals.ts`. Since multiple tools now need to translate sentinels to `null` before calling the API, consider moving it to a shared file like `src/utils/schema-helpers.ts`.
- **[P3] src/tools/add-sections.ts:16**: Repo guidance for new tool features also requires updating the LLM usage instructions in `src/mcp-server.ts`. This change adds section descriptions to the tool surface, but the server prompt still only describes section tools as generic organization helpers, so the new `description` capability isn’t surfaced there.

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-mcp&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-mcp%2Fpull%2F501&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253Af22748b0-6517-11f1-903c-0b5f8199961f%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3Af22748b0-6517-11f1-903c-0b5f8199961f&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/tools/update-sections.ts (hunk 3 above)
Comment (doistbot) [inline src/tools/update-sections.ts]: <big><kbd>🟡 P2</kbd></big>  Making `name` optional means the schema now accepts inputs like `{ id: 'sec-1' }` with no fields to update. Because `execute` casts the request (bypassing the SDK's `RequireAtLeastOne` constraint), this skips validation and calls `client.updateSection(id, {})`. Add a `.refine(data => data.name !== undefined || data.description !== undefined)` to `SectionUpdateSchema` (or add skip logic in `execute`) to prevent this, and include test coverage for the empty-update scenario.

File: src/utils/output-schemas.ts [hunk 5]
```diff
@@ -70,16 +70,22 @@ const ProjectSchema = z.object({
...
- * Strip an SDK Section (or any object with id/name) down to the fields
- * declared in SectionSchema. Keeps tool responses aligned with the schema.
+ * Strip an SDK Section down to the fields declared in SectionSchema. Keeps tool
+ * responses aligned with the schema. The output schema uses an optional string
+ * (Gemini-compatible), so the read's `string | null` description maps `null` to
+ * `undefined`.
  */
-function toSectionSummary({ id, name }: Section): SectionSummary {
-    return { id, name }
+function toSectionSummary({ id, name, description }: Section): SectionSummary {
```
Comment (doistbot) [inline src/utils/output-schemas.ts:88]: <big><kbd>🟡 P2</kbd></big>  This helper updates add/update/get-overview, but `find-sections` still hand-builds `{ id, name }` results. After the 10.4.0 SDK bump, a `find-sections` call on a section that has a description will still drop `description`, so the shared section output surface stays inconsistent. Reuse `toSectionSummary` there as well (or keep `description` out of the shared schema until all section-returning tools populate it).

File: src/tools/__tests__/fetch-object.test.ts [hunk 6]
```diff
@@ -19,6 +19,7 @@ const { FETCH_OBJECT } = ToolNames
 const MOCK_SECTION: Section = {
     id: 'section123',
     name: 'My Section',
+    description: null,
```
Comment (doistbot) [inline src/tools/__tests__/fetch-object.test.ts]: <big><kbd>🟡 P2</kbd></big>  This fixture update only satisfies the new SDK type; it doesn't exercise the new contract that section outputs can include `description`. Set `MOCK_SECTION.description` to a real value and assert it appears in `structuredContent.object`, otherwise `fetch-object` can keep dropping the field with no test failure.

**Review by doistbot — COMMENTED** [review]
This introduces section descriptions to the write tools while cleanly navigating the current SDK limitations.

Few things worth tightening:
- Prevent empty API calls in `update-sections` by adding a schema `.refine()` or skip logic when no fields are provided, alongside an empty-update test.
- Reuse `toSectionSummary` in `find-sections` so it doesn't continue hand-building results and dropping the description field.
- Update the mock section fixtures with a real description value to explicitly assert it correctly surfaces in the tool output.

I also included a few optional follow-up notes in the details below.

<details>
<summary>Optional follow-up notes (2)</summary>

- **[P3] src/tools/add-sections.ts:16**: Per `AGENTS.md`, new tool features must be documented in the LLM usage instructions in `src/mcp-server.ts`. This diff adds section descriptions to the schema, but the server prompt likely needs an update to advertise this capability to the LLM.
- **[P3] src/tools/update-sections.ts:25**: This duplicates the `mapSentinelToNull` helper from `update-goals.ts`. Since multiple tools now translate sentinels to `null` before calling the API, consider extracting it to a shared file like `src/utils/schema-helpers.ts`.

</details>

<sub>[Share Feedback](https://github.com/Doist/doistbot-app/issues/new?template=review-feedback.yml&title=Review+feedback+for+Doist%2Ftodoist-mcp&pr_url=https%3A%2F%2Fgithub.com%2FDoist%2Ftodoist-mcp%2Fpull%2F501&datadog_url=https%3A%2F%2Fapp.datadoghq.com%2Flogs%3Fquery%3Dservice%253A%2528doistbot-app-sandbox%2520OR%2520doistbot-app%2529%2520%2540meta.deliveryId%253A54ad8960-6589-11f1-8bf4-e62855111c44%26agg_m%3Dcount%26agg_m_source%3Dbase%26agg_t%3Dcount%26clustering_pattern_field_path%3Dmessage%26cols%3Dhost%252Cservice%26fromUser%3Dtrue%26messageDisplay%3Dinline%26refresh_mode%3Dsliding%26storage%3Dhot%26stream_sort%3Ddesc%26viz%3Dstream%26live%3Dtrue) • [Review Logs](https://app.datadoghq.com/logs?query=service%3A%28doistbot-app-sandbox%20OR%20doistbot-app%29%20%40meta.deliveryId%3A54ad8960-6589-11f1-8bf4-e62855111c44&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&live=true)</sub>

File: src/tools/update-sections.ts (hunk 3 above)
Comment (lmjabreu) [reply src/tools/update-sections.ts]: Fixed. A `.refine` requires at least one of name/description, so `{ id }` alone is a validation error. In 6af6977.

Comment (lmjabreu) [reply src/tools/update-sections.ts]: Fixed. In 6af6977 (schema-level `.refine`); the no-op input never reaches the SDK cast.

File: src/tools/__tests__/update-sections.test.ts (hunk 4 above)
Comment (lmjabreu) [reply src/tools/__tests__/update-sections.test.ts:72]: Done. Added the `{ id }`-rejected test in 6af6977.

File: src/utils/output-schemas.ts (hunk 5 above)
Comment (lmjabreu) [reply src/utils/output-schemas.ts:88]: Fixed. `find-sections` reuses `toSectionSummary` now, so `description` is included. In 6af6977.

File: src/tools/__tests__/fetch-object.test.ts (hunk 6 above)
Comment (lmjabreu) [reply src/tools/__tests__/fetch-object.test.ts]: Done. `MOCK_SECTION.description` is a real value and the test asserts it appears in `structuredContent.object`; `fetch-object` also uses `toSectionSummary` now. In 6af6977.

File: src/tools/__tests__/update-sections.test.ts (hunk 1 above)
Comment (lmjabreu) [reply src/tools/__tests__/update-sections.test.ts:72]: Done. `"remove"` and legacy `null` clear tests, plus a `""` rejection test; sections share the same `"remove"`-only contract as the projects PR now. In 6af6977.

File: src/tools/update-sections.ts (hunk 2 above)
Comment (lmjabreu) [reply src/tools/update-sections.ts]: Done. Sections implement the `"remove"` sentinel contract (legacy `null` preprocessed, `""` rejected). In 6af6977.

File: src/tools/update-sections.ts [hunk 7]
```diff
@@ -4,10 +4,35 @@ import type { TodoistTool } from '../todoist-tool.js'
 import { SectionSchema as SectionOutputSchema, toSectionSummary } from '../utils/output-schemas.js'
 import { ToolNames } from '../utils/tool-names.js'
 
-const SectionUpdateSchema = z.object({
-    id: z.string().min(1).describe('The ID of the section to update.'),
-    name: z.string().min(1).describe('The new name of the section.'),
-})
+const REMOVE_SENTINEL = 'remove'
```
Comment (gnapse) [inline src/tools/update-sections.ts]: Same as what I mentioned in my review of #500: using a certain string as the signal to unset the description is not acceptable. It prevents calls that genuinely want to set that particular string as the description.

**Review by gnapse — DISMISSED** [review]

**Comment by lmjabreu** [comment]
Unassigning the review for now, same as #500: these are on hold until the feature releases. The sentinel removal agreed on #500 is being applied here too for parity; I'll re-request once the hold lifts.

File: src/tools/update-sections.ts (hunk 7 above)
Comment (lmjabreu) [reply src/tools/update-sections.ts]: Fixed. Same as #500 (replaced the "remove" sentinel with null).

**Comment by lmjabreu** [comment]
> Unassigning the review for now, same as #500: these are on hold until the feature releases.

The squad (Dom) decided it's okay to proceed with the CLI/MCP. All feedback has been addressed in the meanwhile, re-requesting review 👏

**Comment by lmjabreu** [comment]
@gnapse Would you be able to review this when you have a moment? Descriptions is already on web and mobile internally 😊 No rush as we can release without this.

**Review by scottlovegrove — APPROVED** [review]
